crates/services/txpool_v2/src/service.rs

```diff
@@ -333,33 +313,27 @@ where
         }
 
         // Remove expired transactions
-        let mut removed_txs = vec![];
-        {
-            let mut height_expiration_txs = self.pruner.height_expiration_txs.write();
-            let range_to_remove = height_expiration_txs
-                .range(..=new_height)
-                .map(|(k, _)| *k)
-                .collect::<Vec<_>>();
-            for height in range_to_remove {
-                let expired_txs = height_expiration_txs.remove(&height);
-                if let Some(expired_txs) = expired_txs {
-                    let mut tx_pool = self.pool.write();
-                    removed_txs
-                        .extend(tx_pool.remove_transaction_and_dependents(expired_txs));
+        let range_to_remove = self
+            .pruner
+            .height_expiration_txs
+            .range(..=new_height)
+            .map(|(k, _)| *k)
+            .collect::<Vec<_>>();
```

Comment:
do we need this `collect`?

the below loop should work without it too?

Reply:
It will not because iterator has read access to the `self.pruner.height_expiration_txs`